messages/src/msg_fields/protocols/cred_issuance/mod.rs

```diff
@@ -142,22 +142,31 @@ impl Serialize for CredentialPreviewMsgType {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum AttrValue {
+    Plain {
+        value: String,
+    },
+    Encoded {
+        value: String,
+        #[serde(rename = "mime-type")]
+        mime_type: MaybeKnown<MimeType>,
+    },
+}
+
```

Comment:
While not important, I'm honestly not a big fan of enum variants with fields. I feel like a struct can provide better functionality at the cost of being slightly more verbose.

Also, the deserialization would have to be untagged. With that in mind, the `Encoded` variant would have to be attempted first, as the `Plain` variant should always get successfully deserialized because there's always going to be a `value` field (assuming the input is right).